source/h1_decoder.c

```diff
@@ -411,6 +414,14 @@ static int s_linestate_header(struct aws_h1_decoder *decoder, struct aws_byte_cu
 
     switch (header.name) {
         case AWS_HTTP_HEADER_CONTENT_LENGTH:
+            if (decoder->content_length_received) {
+                AWS_LOGF_ERROR(
+                    AWS_LS_HTTP_STREAM,
+                    "id=%p: Multiple incoming headers for content-length received. This is illegal.",
```

Comment:
As I [mentioned](https://github.com/awslabs/aws-c-http/discussions/493#discussioncomment-11378308) a few months back, the server stuff was in an experimental state. Not something we'd recommend. We never really did a top-to-bottom read of the RFC when implementing HTTP/1. There are very very likely more security issues like this that would affect a server, which needs to worry about receiving malicious requests. At least, much more than a client needs to worry while receiving responses from AWS, which is what this library was primarily developed for.